### PR TITLE
AP_BattMonitor: Use a common PEC helper

### PR DESCRIPTION
**Pull request by WickedShell**
PEC is part of the SMBus spec, it won't change between backends, it makes more sense to use a single helper for it. Also saves 112 bytes of flash :)

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include "AP_BattMonitor_SMBus.h"
+
+#define AP_BATTMONITOR_SMBUS_PEC_POLYNOME 0x07 // Polynome for CRC generation
+
+/// get_PEC - calculate packet error correction code of buffer
+uint8_t AP_BattMonitor_SMBus::get_PEC(const uint8_t i2c_addr, uint8_t cmd, bool reading, const uint8_t buff[], uint8_t len) const
+{
+    // exit immediately if no data
+    if (len <= 0) {
```
Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: `len == 0`, it can't be less than 0.

Comment (WickedShell) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Preexsisting error :)

Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Actually was fixed in Maxell version :stuck_out_tongue_winking_eye:

Comment (WickedShell) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: Yeah this is why they need to be unified, so stuff like that stops happening :)

Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: GitHub only allows me to put one +1, otherwise I'd put a lot more :slightly_smiling_face:

**Comment by OXINARF** [comment]
Merged with #6091

**Pull request closed without merging**